libnm-core/nm-utils.c

```diff
@@ -2455,21 +2482,45 @@ nm_utils_tc_qdisc_from_str (const char *str, GError **error)
 	gs_free char *kind = NULL;
 	gs_free char *rest = NULL;
 	NMTCQdisc *qdisc = NULL;
-	gs_unref_hashtable GHashTable *ht = NULL;
+	gs_unref_hashtable GHashTable *options = NULL;
+	GHashTableIter iter;
+	gpointer key, value;
+	int i;
 
 	nm_assert (str);
 	nm_assert (!error || !*error);
 
-	ht = nm_utils_parse_variant_attributes (str,
+	options = nm_utils_parse_variant_attributes (str,
 	                                        ' ', ' ', FALSE,
 	                                        tc_object_attribute_spec,
 	                                        error);
-	if (!ht)
+	if (!options)
 		return NULL;
+	g_clear_pointer (&options, g_hash_table_unref);
 
 	if (!_tc_read_common_opts (str, &handle, &parent, &kind, &rest, error))
 		return NULL;
 
+	for (i = 0; rest && tc_qdisc_attribute_spec[i]; i++) {
+		if (strcmp (tc_qdisc_attribute_spec[i]->kind, kind) == 0) {
+			options = nm_utils_parse_variant_attributes (rest,
+			                                             ' ', ' ', FALSE,
+			                                             tc_qdisc_attribute_spec[i]->attrs,
+			                                             error);
+			if (!options)
+				return NULL;
+			break;
+		}
+	}
+
+	if (options) {
+		value = g_hash_table_lookup (options, "");
+		if (value)
+			rest = g_variant_dup_string (value, NULL);
+		else
+			rest = NULL;
```

Comment:
this looks not right. What does it do? What's special about `""`?

Also, the ownership handling of `@rest` seems not right. Above, I'd expect `rest` to own some string. Here, you just reset the value.